pkg/vpcmodel/explainabilityPrint.go

```diff
@@ -129,11 +129,13 @@ func (g *groupedConnLine) explainabilityLineStr(c *VPCConfig, connQuery *connect
 	loadBalancerBlocking := loadBalancerRule != nil && loadBalancerRule.Deny(false)
 	ingressBlocking := !expDetails.ingressEnabled && needIngress
 	egressBlocking := !expDetails.egressEnabled && needEgress
+	isExternal := src.IsExternal() || dst.IsExternal()
 	var externalRouterHeader, crossRouterFilterHeader, loadBalancerHeader, resourceEffectHeader,
 		crossRouterFilterDetails, loadBalancerDetails, details string
 	externalRouter, crossVpcRouter, crossVpcRules := expDetails.externalRouter, expDetails.crossVpcRouter, expDetails.crossVpcRules
+	missingExternalRouter := isExternal && externalRouter == nil
```

Comment:
Move this line closer to where the variable is actually used

Reply:
done